File: javascript/gpt/p123.js
// Check that exactly one argument is provided
if (process.argv.length !== 3) {
    console.error(`Usage: node script.js <string>`);
    process.exit(1);
}

let str = process.argv[2];

// Ensure argument is a string
if (typeof str !== "string") {
    console.error("Invalid string");
    process.exit(1);
}

// Trim trailing whitespace
str = str.replace(/\s+$/, "");

// Trim trailing newlines
str = str.replace(/[\r\n]+$/, "");

console.log(`Trimmed string: "${str}"`);

File: python/gpt/p123.py
import sys

# Check that exactly one argument is provided
if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} <string>")
    sys.exit(1)

arg = sys.argv[1]

# Ensure the argument is a string
if not isinstance(arg, str):
    print("Invalid string")
    sys.exit(1)

# Trim trailing whitespace
arg = arg.rstrip()

# Trim trailing newlines
arg = arg.rstrip("\n\r")

print(f'Trimmed string: "{arg}"')
